File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlSetValueStatement.cs
namespace T1.SqlSharp.Expressions;

public class SqlSetValueStatement : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.SetValueStatement;
    public required ISqlExpression Name { get; set; } = new SqlValue();
    public string Operator { get; set; } = "=";
    public required ISqlExpression Value { get; set; } = new SqlValue();
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_SetValueStatement(this);
    }

    public string ToSql()
    {
        return $"SET {Name.ToSql()} {Operator} {Value.ToSql()}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlSetValueStatement.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlSetValueStatement implements ISqlExpression {
    SqlType: SqlType = SqlType.SetValueStatement;
    Span: TextSpan = new TextSpan();
    Name!: ISqlExpression;
    Value!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_SetValueStatement(this);
    }

    ToSql(): string {
        return `${this.Name.ToSql()} = ${this.Value.ToSql()}`;
    }
} 